src/NuGet.Core/NuGet.Commands/RestoreCommand/RestoreRunner.cs

```diff
@@ -41,7 +44,34 @@ public static class RestoreRunner
             return await RunAsync(restoreContext, CancellationToken.None);
         }
 
-        /// <summary>
+        private static async Task SetDefaultCredentialProviderAsync(ILogger logger)
+        {
+            if (HttpHandlerResourceV3.CredentialService == null)
+            {
+                HttpHandlerResourceV3.CredentialService = new CredentialService(await GetCredentialProvidersAsync(logger), nonInteractive: false);
+            }
+        }
+
+        // Add only the secure plugin. This will be done when there's nothing set
+        private static async Task<IEnumerable<Credentials.ICredentialProvider>> GetCredentialProvidersAsync(ILogger logger )
+        {
+            var providers = new List<Credentials.ICredentialProvider>();
+
+            var securePluginProviders = await (new SecureCredentialProviderBuilder(PluginManager.Instance, logger)).BuildAll();
+
+            providers.AddRange(securePluginProviders);
+
+            if (securePluginProviders.Any())
+            {
+                if (PreviewFeatureSettings.DefaultCredentialsAfterCredentialProviders)
+                {
+                    providers.Add(new DefaultCredentialsCredentialProvider());
```

Comment:
this `DefaultCredentialsCredentialProvider` is so confusing... I know it's existing impl but shouldn't we just change it to something more meaningful like `DefaultNetworkCredentialProvider`

Reply:
I'll change it in a subsequent PR.